analyzer/pkg/dpkg/dpkg.go

```diff
@@ -29,6 +30,19 @@ type debianPkgAnalyzer struct{}
 func (a debianPkgAnalyzer) Analyze(fileMap extractor.FileMap) (pkgs []analyzer.Package, err error) {
 	detected := false
 	for _, filename := range a.RequiredFiles() {
+		// find by target directory
+		if filename[len(filename)-1:] == "/" {
+			for targetname, targetbytes := range fileMap {
```

Comment:
It seems better to make this loop the outer.

Before
```
for _, filename := range a.RequiredFiles()
    for targetname, targetbytes := range fileMap
```

After
```
for targetName, targetBytes := range fileMap
    dir := filepath.Dir(targetName)
    if !utils.StringInSlice(targetName, a.requiredFiles()) && !utils.StringInSlice(dir, a.requiredDirs()){
        continue
    }
```

Reply:
Currently, I added "/" to last char.

```
dir := filepath.Dir(filename) + "/"
```

If you don't like, I can create RequiredDir func.
check => https://github.com/aquasecurity/fanal/pull/29#discussion_r320302883